src/NHibernate/Linq/GroupBy/PagingRewriter.cs

```diff
@@ -46,9 +45,13 @@ private static void FlattenSubQuery(SubQueryExpression subQueryExpression, Query
 			{
 				var cro = new ContainsResultOperator(new QuerySourceReferenceExpression(subQueryMainFromClause));
 
+				// Cloning may cause having/join/with clauses listed in VisitorParameters to no more be matched.
+				// Not a problem for now, because those clauses imply a projection, which is not supported
+				// by the "new WhereClause(new SubQueryExpression(newSubQueryModel))" below. See
+				// SingleKeyGroupAndCountWithHavingClausePagingAndOuterWhere test by example.
```

Comment:
This assertion was tested by adding following test to `NHibernate.Test.Linq.ByMethod.GroupByHavingTests`:

```C#
[Test, Explicit("Demonstrate an unsupported case for PagingRewriter")]
public void SingleKeyGroupAndCountWithHavingClausePagingAndOuterWhere()
{
	var orderCounts = db.Orders
		.GroupBy(o => o.Customer.CompanyName)
		.Where(g => g.Count() > 10)
		.Select(g => new { CompanyName = g.Key, OrderCount = g.Count() })
		.OrderBy(oc => oc.CompanyName)
		.Skip(5)
		.Take(10)
		.Where(oc => oc.CompanyName.Contains("F"))
		.ToList();

	Assert.That(orderCounts, Has.Count.EqualTo(3));
	var frankRow = orderCounts.Single(row => row.CompanyName == "Frankenversand");
	Assert.That(frankRow.OrderCount, Is.EqualTo(15));
}
```
The `WhereClause` line yields an `ArgumentException` with message

```
The items of the input sequence of type '<>f__AnonymousType6`2[System.String,System.Int32]' are not
compatible with the item expression of type
'System.Linq.IGrouping`2[System.String,NHibernate.DomainModel.Northwind.Entities.Order]'.
```

Which lets me think this flattening process does not support projection (the `Select(g =>`), which are needed for introducing having or left join clauses.

I have included that test case in this PR just now.